.github/workflows/ci-workflow.yml

```diff
@@ -1,58 +1,68 @@
-name: E4E Pipeline
-
-on: [push, pull_request]
-
+name: E4E CI Workflow
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches:
+      - "*"
+env:
+  PROJECT_NAME: explorer-for-endevor
 jobs:
-  build:
+  build-and-test:
     runs-on: ${{ matrix.os }}
-
-    env:
-      PROJECT_NAME: explorer-for-endevor
-
     strategy:
       # Continue to run tests on the other systems if one fails
       fail-fast: false
       matrix:
         node-version: [12.x, 14.x, 16.x]
         os: [windows-latest, ubuntu-latest, macos-latest]
-
     steps:
       - name: Check out code
         uses: actions/checkout@v2
-
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v1
         with:
           node-version: ${{ matrix.node-version }}
-
       - name: Install dependencies
         run: yarn install --frozen-lockfile
-
       - name: Audit
+        run: yarn audit
         id: audit
         continue-on-error: true
-        run: yarn audit
-
       - name: Lint
         run: yarn lint
-
       - name: Run tests
         uses: GabrielBB/xvfb-action@v1.0
+        id: tests
         with:
           run: yarn test
+      - name: Check exit status code
+        if: steps.tests.outputs.exit_status < 7 && steps.audit.outputs.exit_status < 7
+        run: exit 0
 
-      - name: Build package
-        if: steps.audit.outputs.exit_code < 7
+  create-artifact:
+    runs-on: windows-latest
+    needs: build-and-test
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Get E4E package version
```

Comment:
I want to add the version number onto the VSIX file. We can get the VSIX out of the folder with a wildcard, but then it will be uploaded everywhere as `explorer-for-endevor.vsix` and not `explorer-for-endevor-1.0.12.vsix`. I think it's important for people to be able to tell which version of VSIX they have downloaded, before they install it in VSCode from the VSIX.

And IMHO it's not so bad to leave this step in there. It is getting the version from the `package.json` file, where the version number will always be. If we move the `package.json` file, it means we are changing the structure of the whole repo, and we will have much bigger problems than adjusting this path (which is easy to do) if we end up changing the repo structure.

Reply:
Yep, that make sense, I agree with you.